Crossplane pod crashes when debug is enabled

### What happened?
The crossplane pod crashes when debug is enabled because there are log.Debug statements that have the wrong number of arguments:

```
1.651772500306353e+09	DPANIC	crossplane	odd number of arguments passed as key-value pairs for logging	{"controller": "defined/compositeresourcedefinition.apiextensions.crossplane.io", "request": "/xeksns.saas.nokia.com", "uid": "20fac691-2107-4ce8-9530-3d9309733301", "version": "641224762", "name": "xeksns.saas.nokia.com", "controller": "composite/xeksns.saas.nokia.com", "request": "/bobtestpkt-n98vg-bdnjl", "uid": "36a4ab20-0ec3-4f35-bc03-f7b57881673f", "version": "641262182", "name": "bobtestpkt-n98vg-bdnjl", "deletion-timestamp": "2022-05-05 16:35:33 +0000 UTC", "ignored key": "error"}
github.com/crossplane/crossplane/internal/controller/apiextensions/composite.(*Reconciler).Reconcile
	/home/runner/work/crossplane/crossplane/internal/controller/apiextensions/composite/reconciler.go:417
github.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile
	/home/runner/work/crossplane/crossplane/vendor/github.com/crossplane/crossplane-runtime/pkg/ratelimiter/reconciler.go:54
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/home/runner/work/crossplane/crossplane/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/home/runner/work/crossplane/crossplane/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/home/runner/work/crossplane/crossplane/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/home/runner/work/crossplane/crossplane/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227
panic: odd number of arguments passed as key-value pairs for logging

goroutine 2105 [running]:
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc000ccd080, {0xc003c82680, 0x1, 0x1})
	/home/runner/work/crossplane/crossplane/vendor/go.uber.org/zap/zapcore/entry.go:232 +0x446
go.uber.org/zap.(*Logger).DPanic(0x1dcf62f, {0x1e31f0d, 0x1a5d180}, {0xc003c82680, 0x1, 0x1})
	/home/runner/work/crossplane/crossplane/vendor/go.uber.org/zap/logger.go:220 +0x59
github.com/go-logr/zapr.(*zapLogger).handleFields(0xc00341cb70, 0x1, {0xc002564b20, 0x1, 0x40adfd}, {0x0, 0x1c93a00, 0x1})
	/home/runner/work/crossplane/crossplane/vendor/github.com/go-logr/zapr/zapr.go:145 +0xdea
github.com/go-logr/zapr.(*zapLogger).Info(0xc00341cb70, 0x1, {0x1e04592, 0x1535f4e}, {0xc002564b20, 0x1, 0x1})
	/home/runner/work/crossplane/crossplane/vendor/github.com/go-logr/zapr/zapr.go:198 +0x8d
github.com/go-logr/logr.Logger.Info({{0x20b96a0, 0xc00341cb70}, 0xc004738280}, {0x1e04592, 0x2a}, {0xc002564b20, 0x1, 0x1})
	/home/runner/work/crossplane/crossplane/vendor/github.com/go-logr/logr/logr.go:252 +0xd0
github.com/crossplane/crossplane-runtime/pkg/logging.logrLogger.Debug(...)
	/home/runner/work/crossplane/crossplane/vendor/github.com/crossplane/crossplane-runtime/pkg/logging/logging.go:89
github.com/crossplane/crossplane/internal/controller/apiextensions/composite.(*Reconciler).Reconcile(0xc003fe7700, {0x209ab00, 0xc004adac30}, {{{0x0, 0xc00222bc30}, {0xc004b85038, 0x30}}})
	/home/runner/work/crossplane/crossplane/internal/controller/apiextensions/composite/reconciler.go:417 +0xad1
github.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile(0xc002db8320, {0x209ab00, 0xc004adac30}, {{{0x0, 0x1ce6ea0}, {0xc004b85038, 0x30}}})
	/home/runner/work/crossplane/crossplane/vendor/github.com/crossplane/crossplane-runtime/pkg/ratelimiter/reconciler.go:54 +0x16b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc00033e4d0, {0x209ab00, 0xc004adab70}, {{{0x0, 0x1ce6ea0}, {0xc004b85038, 0x413ab4}}})
	/home/runner/work/crossplane/crossplane/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:114 +0x26f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00033e4d0, {0x209aa58, 0xc004815ac0}, {0x1bd42a0, 0xc004443de0})
	/home/runner/work/crossplane/crossplane/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:311 +0x33e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00033e4d0, {0x209aa58, 0xc004815ac0})
	/home/runner/work/crossplane/crossplane/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/home/runner/work/crossplane/crossplane/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/home/runner/work/crossplane/crossplane/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:223 +0x357
```

### How can we reproduce it?
I'm not entirely sure what triggered it as I was running with debug enabled for weeks before it started crashing

### What environment did it happen in?
Crossplane version: 1.7

I'll push a PR that fixes the logging errors that I can find.

--- a/internal/controller/apiextensions/claim/reconciler.go
+++ b/internal/controller/apiextensions/claim/reconciler.go
@@ -408,7 +408,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 
 	if err := r.claim.AddFinalizer(ctx, cm); err != nil {
-		log.Debug(errAddFinalizer, "error")
+		log.Debug(errAddFinalizer, "error", err)
 		err = errors.Wrap(err, errAddFinalizer)
 		record.Event(cm, event.Warning(reasonBind, err))
 		return reconcile.Result{}, err

--- a/internal/controller/apiextensions/composite/reconciler.go
+++ b/internal/controller/apiextensions/composite/reconciler.go
@@ -414,7 +414,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		}
 
 		if err := r.composite.RemoveFinalizer(ctx, cr); err != nil {
-			log.Debug(errRemoveFinalizer, "error")
+			log.Debug(errRemoveFinalizer, "error", err)
 			err = errors.Wrap(err, errRemoveFinalizer)
 			r.record.Event(cr, event.Warning(reasonDelete, err))
 			return reconcile.Result{}, err
@@ -425,7 +425,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 
 	if err := r.composite.AddFinalizer(ctx, cr); err != nil {
-		log.Debug(errAddFinalizer, "error")
+		log.Debug(errAddFinalizer, "error", err)
 		err = errors.Wrap(err, errAddFinalizer)
 		r.record.Event(cr, event.Warning(reasonInit, err))
 		return reconcile.Result{}, err

